### PR TITLE
[RUN-3576] Mention required dependencies for custom repositories (Mx 10.5.0)

### PR DESCRIPTION
**Pull request by atennapel**
This PR updates the managed dependencies documentation to mention the platform required dependencies.

### CODE REVIEW

File: content/en/docs/refguide/java-programming/managed-dependencies.md [hunk 1]
```diff
@@ -71,18 +71,30 @@ Platform-supported Marketplace modules created by Mendix have been updated with
...
+
 For more details, refer to the Gradle documentation on [Declaring repositories](https://docs.gradle.org/current/userguide/declaring_repositories.html).
 
 {{< figure src="/attachments/refguide/java-programming/managed-dependencies/custom-repository.png" >}}
 
+### 6.1 Required dependencies{#custom-repos-required-dependencies}
+
+There are some dependencies that are required by Mendix. These need to be added to your configured repository. Below is a list of these dependencies.
+
+1. The Gradle plugin [cyclonedx-gradle-plugin](https://github.com/CycloneDX/cyclonedx-gradle-plugin), used for Software Bill of Material generation.
```
Comment (lvanengelen) [inline content/en/docs/refguide/java-programming/managed-dependencies.md]: 🎄 I wouldn't use a numbered list as the order doesn't matter

File: content/en/docs/refguide/java-programming/managed-dependencies.md [hunk 2]
```diff
@@ -71,18 +71,30 @@ Platform-supported Marketplace modules created by Mendix have been updated with
 
 By default, dependencies are downloaded from the [Maven Central](https://central.sonatype.com/) repository. In some scenarios, you may want to specify a custom location. For example, if your organization has its own repository to cache downloads or as an alternative if internet access is restricted in an air-gapped setup.
 
-Custom repositories are configured in the **Repositories** setting of the **Deployment** tab in the [Preferences](/refguide/preferences-dialog/) dialog box. This setting uses the same syntax as Gradle. For example, to resolve dependencies from a directory `lib`, enter the following:
+Custom repositories are configured in the **Repositories** setting of the **Deployment** tab in the [Preferences](/refguide/preferences-dialog/) dialog box. This setting uses the same syntax as Gradle. The configured repositories are also used to resolve Gradle plugins. For example, to resolve dependencies from a directory `lib`, enter the following:
```
Comment (lvanengelen) [inline content/en/docs/refguide/java-programming/managed-dependencies.md]: As a developer can't these plugins, this sounds a bit magic.

Maybe something like 'For internal usage, Studio Pro uses Gradle plugins which are also resolved from the configured repositories'.

File: content/en/docs/refguide/java-programming/managed-dependencies.md [hunk 3]
```diff
@@ -71,18 +71,30 @@ Platform-supported Marketplace modules created by Mendix have been updated with
...
 
 ```groovy {linenos=false}
 flatDir {
     dirs 'lib'
```
Comment (lvanengelen) [inline content/en/docs/refguide/java-programming/managed-dependencies.md:78]: This example will now almost fail, unless the project has no managed dependencies at all. We should extend the example with `gradlePluginPortal()` and `mavenCentral()`. That might also give a hook to explain them a bit better.

Comment (atennapel) [reply content/en/docs/refguide/java-programming/managed-dependencies.md:78]: But the point of this example is to use a custom repository. This part assumes that you know what you are doing and it links the Gradle repositories documentation for more details.

File: content/en/docs/refguide/java-programming/managed-dependencies.md (hunk 2 above)
Comment (atennapel) [reply content/en/docs/refguide/java-programming/managed-dependencies.md]: Done.

File: content/en/docs/refguide/java-programming/managed-dependencies.md (hunk 1 above)
Comment (atennapel) [reply content/en/docs/refguide/java-programming/managed-dependencies.md]: Done.

File: content/en/docs/refguide/java-programming/managed-dependencies.md [hunk 4]
```diff
@@ -71,18 +71,30 @@ Platform-supported Marketplace modules created by Mendix have been updated with
...
+
 For more details, refer to the Gradle documentation on [Declaring repositories](https://docs.gradle.org/current/userguide/declaring_repositories.html).
 
 {{< figure src="/attachments/refguide/java-programming/managed-dependencies/custom-repository.png" >}}
 
+### 6.1 Required dependencies{#custom-repos-required-dependencies}
+
+There are some dependencies that are required by Mendix. These need to be added to your configured repository. Below is a list of these dependencies.
+
+- The Gradle plugin [cyclonedx-gradle-plugin](https://github.com/CycloneDX/cyclonedx-gradle-plugin), used for Software Bill of Material generation.
```
Comment (lvanengelen) [inline content/en/docs/refguide/java-programming/managed-dependencies.md]: 🎄 bill of material**_s_**

File: content/en/docs/refguide/java-programming/managed-dependencies.md [hunk 5]
```diff
@@ -71,7 +71,7 @@ Platform-supported Marketplace modules created by Mendix have been updated with
 
 By default, dependencies are downloaded from the [Maven Central](https://central.sonatype.com/) repository. In some scenarios, you may want to specify a custom location. For example, if your organization has its own repository to cache downloads or as an alternative if internet access is restricted in an air-gapped setup.
 
-Custom repositories are configured in the **Repositories** setting of the **Deployment** tab in the [Preferences](/refguide/preferences-dialog/) dialog box. This setting uses the same syntax as Gradle. The configured repositories are also used to resolve Gradle plugins. For example, to resolve dependencies from a directory `lib`, enter the following:
+Custom repositories are configured in the **Repositories** setting of the **Deployment** tab in the [Preferences](/refguide/preferences-dialog/) dialog box. This setting uses the same syntax as Gradle. For internal usage, Studio Pro uses Gradle plugins which are also resolved using the configured repositories. For example, to resolve dependencies from a directory `lib`, enter the following:
```
Comment (lvanengelen) [inline content/en/docs/refguide/java-programming/managed-dependencies.md]: Maybe keep this generic, not specific to Gradle plugins (like the error message in SP)? Something like: 'For internal usage of the platform, some dependencies are required which are also resolved using the configured repositories.'

Comment (atennapel) [reply content/en/docs/refguide/java-programming/managed-dependencies.md]: Yes that's better. Done.

File: content/en/docs/refguide/java-programming/managed-dependencies.md [hunk 6]
```diff
@@ -105,21 +105,25 @@ If you have an issue with the managed dependencies of a Marketplace module, you
...
 
-3. CE9805 – Network connection failure.
+4. CE9807 - Required dependency could not be found in the configured repository.
 
-    * Check that you have a working internet connection.
-    * If you are using a [custom repository](#custom-repos), check that the repository can be reached.
+    * Ensure that all the [required dependencies](#custom-repos-required-dependencies) exist in the configured repository. 
 
-4. CE9803 – Any failure which is not covered in the above scenarios.
+5. CE9803 – Any failure which is not covered in the above scenarios.
```
Comment (lvanengelen) [inline content/en/docs/refguide/java-programming/managed-dependencies.md:126]: If the goal is to sort on error code number, this one is still out of order 😄

**Comment by atennapel** [comment]
@ConnorLand This feature is delayed to 10.5.0.

**Review by ConnorLand — APPROVED** [review]

**Pull request merged**